### DISCUSSION TITLE
Using Both JS and GO api's

### DISCUSSION THREAD

**Discussion by LiorStrugach** [pocketbase/pocketbase#5220] [Q&A]
Hey, I have an already written out database creation scripts that I created in JS using the pocketbase goja plugin. Now I am planning to add some external DB calls and I want to implement them using the native go library.

Is there a way to use the hooks created using the goja plugin in conjunction with the external capabilities I am going to add in go?

**Comment by ganigeorgiev** [comment]
Yes, you can load the JSVM `pb_hooks` and `pb_migrations` in similar manner to the prebuilt [example/main.go](https://github.com/pocketbase/pocketbase/blob/966341117193f8060fd85c5d57af92651ce3b1aa/examples/base/main.go#L96C1-L102C1) file, for example:

```go
jsvm.MustRegister(app, jsvm.Config{
    WatchHooks:    true,
    HooksPoolSize: 15,
    OnInit: func(vm *goja.Runtime) {
        // register custom bindings
        vm.Set("myCustomVar", 123)
    },
})
```

More info about the JSVM config options you can find in the [`jsvm` godoc](https://pkg.go.dev/github.com/pocketbase/pocketbase@v0.22.17/plugins/jsvm).
